internal/controller/backup/reconcile.go

```diff
@@ -96,8 +109,76 @@ func (r *Reconciler) ReconcileBackup(ctx context.Context, mgmtBackup *kcmv1alpha
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) updateAfterRestoration(ctx context.Context, mgmtBackup *kcmv1alpha1.ManagementBackup) (ctrl.Result, error) {
+	removeVeleroLabels := func() {
+		delete(mgmtBackup.Labels, velerov1.BackupNameLabel)
+		delete(mgmtBackup.Labels, velerov1.RestoreNameLabel)
+	}
+
+	l := ctrl.LoggerFrom(ctx).V(1)
+
+	if mgmtBackup.Status.LastBackup != nil || mgmtBackup.Status.LastBackupName != "" || !mgmtBackup.Status.LastBackupTime.IsZero() { // fast-track
+		l.Info("Removing velero labels after restoration when status is already set")
+		removeVeleroLabels()
+		if err := r.cl.Update(ctx, mgmtBackup); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update ManagementBackup labels after restoration: %w", err)
+		}
+
+		return ctrl.Result{}, nil
+	}
+
+	veleroBackups := new(velerov1.BackupList)
+	if err := r.cl.List(ctx, veleroBackups); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to list velero Backups: %w", err)
+	}
+
+	updateStatus := false
+	if mgmtBackup.IsSchedule() {
+		l.Info("Updating schedule after restoration")
+		lastBackup, ok := getMostRecentProducedBackup(mgmtBackup.Name, veleroBackups.Items)
+		if ok { // if have not found then there were no backups yet
+			l.Info("Found last backup", "last_backup_name", lastBackup.Name)
+			// next attempt will be fetch on the next event
+			mgmtBackup.Status.LastBackup = &lastBackup.Status
+			mgmtBackup.Status.LastBackupName = lastBackup.Name
+			mgmtBackup.Status.LastBackupTime = lastBackup.Status.StartTimestamp
+			updateStatus = true
+		} else {
+			l.Info("No last backup has been found")
+		}
+	} else {
+		l.Info("Updating single after restoration")
```

Comment:
Let's make it a little bit clearer. For example:
```suggestion
		l.Info("Updating single backup after restoration")
```